### DISCUSSION TITLE
Invoking an Authentication Service

### DISCUSSION THREAD

**Discussion by MasterThreadInf** [unoplatform/uno#10532] [Q&A]
https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Tutorials/Authentication/HowTo-Authentication.html#2-invoking-an-authentication-service

First of all ,where to add POCO is unclear and many things as per documetation causes error. 
for e.g
1. `<Button Content="Login" Click="{x:Bind ViewModel.Authenticate}" />`

and when I try 'Command= {Binding Authenticate}' then it works 

2. In `ShellViewModel`  ,there is no _auth defined 
3. ```
    .UseAuthentication(auth` =>
      auth.AddCustom<IDummyJsonEndpoint>(custom =>
        custom.Login(
          async (authService, dispatcher, tokenCache, credentials, cancellationToken) =>
          {
            var name = credentials.FirstOrDefault(x => x.Key == "Username").Value;
            var password = credentials.FirstOrDefault(x => x.Key == "Password").Value;
            var creds = new Credentials { Username = name, Password = password };
            var authResponse = await authService.Login(creds, cancellationToken);
            if (authResponse?.Token is not null)
            {
              credentials["AccessToken"] = authResponse.Token;
              return credentials;
            }
            return default;
          }));
    ```
This isn't working for wasm. So How do I invoke this for wasm? Or Am I missing something here? 

The document is helpful, but unfortunately there are incomplete informations and mistakes, which takes unnecessary time to resolve it more than it actually needed.  Uno documents really needs update that is what I'm experiencing right now.  Wish there was button like wikipedia, where we can submit correction and after reviewing the correction page will be updated.

**Comment by jeromelaban** [comment]
Thanks for the report. You can see the "Improve this doc" on the right:
![](https://user-images.githubusercontent.com/5839577/203349018-f40a216b-e55f-420a-8b26-fe20cd8f7542.png)

@carldebilly do you have insights here?

**Reply by carldebilly** [reply]
@nickrandolph

**Comment by nickrandolph** [comment]
We're aiming to have docs for authentication updated in next week or so. In meantime, the TestHarness in the uno.extensions repository has some examples of authentication

**Reply by MasterThreadInf** [reply]
link please for the example you're talking about

**Reply by nickrandolph** [reply]
Start here: https://github.com/unoplatform/uno.extensions/tree/main/testing/TestHarness/TestHarness.Shared/Ext/Authentication
